### PR TITLE
em-690 have all unit tests pass

### PR DESCRIPTION
**Pull request by btzc**
- Have the base unit test that angular automatically generates pass for each spec file.
- Write full unit test coverage for news service spec
- Remove protractor and e2e
- Bump Karma version to 1.7.1 and include more dependencies for tests

### CODE REVIEW

File: src/app/compliance-oversight/compliance-oversight.component.spec.ts [hunk 1]
```diff
@@ -1,17 +1,19 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
 import { ComplianceOversightComponent } from './compliance-oversight.component';
 
+import { Jasmine } from 'jasmine-core';
```
Comment (NickPhura) [inline src/app/compliance-oversight/compliance-oversight.component.spec.ts]: I dont think we need this.

File: src/app/services/news.service.spec.ts [hunk 2]
```diff
@@ -1,15 +1,679 @@
...
+        ];
+
+        const sorted = newsService.comparePinned(mockResponse[0], mockResponse[1]);
+        expect(sorted).toBe(-3600000);
+      }));
+      it('returns b is more recent than a',
+          inject([NewsService], (newsService) => {
+
+        const mockResponse = [
+          createNewsItem('Test1', true, 1, '2018-10-14 17:00:00.000Z'),
```
Comment (NickPhura) [inline src/app/services/news.service.spec.ts]: try making other values different,not just year.

**Comment by btzc** [comment]
@esune I tested it with some of the services by just changing random functionality and it broke the tests. But I didn't test with all of them so I can't say for certain.

File: package.json [hunk 3]
```diff
@@ -31,6 +31,8 @@
     "bootstrap": "4.0.0-alpha.6",
     "core-js": "2.4.1",
     "jquery": "3.1.1",
+    "karma": "1.7.1",
```
[the remark this answers is not in the document]
Comment (btzc) [reply package.json]: The number of times i had to sing the alphabet in my head to abide by this request was honestly just too many

File: src/app/compliance-oversight/compliance-oversight.component.spec.ts (hunk 1 above)
Comment (btzc) [reply src/app/compliance-oversight/compliance-oversight.component.spec.ts]: Fixed

File: src/app/services/news.service.spec.ts (hunk 2 above)
Comment (btzc) [reply src/app/services/news.service.spec.ts]: Fixed

File: .angular-cli.json [hunk 4]
```diff
@@ -35,20 +35,12 @@
       }
     }
   ],
-  "e2e": {
```
Comment (asanchezr) [inline .angular-cli.json]: Why did Protractor get removed? Are we using a different solution for e2e tests? Please clarify

File: package.json [hunk 5]
```diff
@@ -52,13 +52,13 @@
...
     "jasmine-core": "~2.5.2",
     "jasmine-spec-reporter": "~3.2.0",
-    "karma": "~1.4.1",
+    "karma": "~1.7.1",
     "karma-chrome-launcher": "~2.2.0",
     "karma-cli": "~1.0.1",
     "karma-coverage-istanbul-reporter": "^0.2.0",
     "karma-jasmine": "~1.1.0",
     "karma-jasmine-html-reporter": "^0.2.2",
-    "protractor": "~5.1.0",
```
Comment (asanchezr) [inline package.json]: Why did Protractor get removed? Are we using a different solution for e2e tests? Please clarify

File: protractor.conf.js [hunk 6]
```diff
@@ -1,30 +0,0 @@
-// Protractor configuration file, see link for more information
```
Comment (asanchezr) [inline protractor.conf.js]: Why did Protractor get removed? Are we using a different solution for e2e tests? Please clarify

**Pull request merged**